### PR TITLE
[DOC] Add core compatibility guide

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Adding a new guide, to show how different ESP32 Arduino core versions can be handled within libraries or projects to keep the compatibility.

### CODE REVIEW

**Review by lucasssvaz — APPROVED** [review]
LGTM

File: docs/en/guides/core_compatibility.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Welcome to the compatibility guide for library developers aiming to support multiple versions of the ESP32 Arduino core. This documentation provides essential tips and best practices for ensuring compatibility with 2.x, 3.x and future versions of the ESP32 Arduino core.
+
+Code Adaptations
+----------------
+
+To ensure compatibility with both versions of the ESP32 Arduino core, developers should utilize conditional compilation directives in their code. Below is an example of how to conditionally include code based on the ESP32 Arduino core version::
+
+    .. code-block:: cpp
+
+      #if ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(3, 0, 0)
```
Comment (me-no-dev) [inline docs/en/guides/core_compatibility.rst:16]: If not defined, you get:
```
error: function-like macro 'ESP_ARDUINO_VERSION_VAL' is not defined
#if ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(3, 0, 0)
                           ^
```

Comment (me-no-dev) [reply docs/en/guides/core_compatibility.rst:16]:
```cpp
#ifdef ESP_ARDUINO_VERSION_MAJOR
#if ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(3, 0, 0)
    printf("Version 3.x\n");
#else
    printf("Version 2.x\n");
#endif
#else
    printf("Version 1.x\n");
#endif
```

Comment (P-R-O-C-H-Y) [reply docs/en/guides/core_compatibility.rst:17]: Updated

File: docs/en/guides/core_compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
+Compatibility Guide for ESP32 Arduino Core
+====================================================
```
Comment (pedrominatel) [inline docs/en/guides/core_compatibility.rst:2]: Remove extra chars.

**Review by pedrominatel — APPROVED** [review]
LGTM

**Review by me-no-dev — APPROVED** [review]

File: docs/en/guides/core_compatibility.rst (hunk 2 above)
Comment (P-R-O-C-H-Y) [reply docs/en/guides/core_compatibility.rst]: Done, btw. @pedrominatel why this did not trigger any warning or error?

**Pull request merged**